dojo/tools/auditjs/parser.py

```diff
@@ -0,0 +1,125 @@
+            if "coordinates" in dependency:
+                file_path = dependency["coordinates"]
+                component_name, component_version = file_path.split('/')[1].split('@')
+
+            # Check if there are any vulnerabilities
+            if dependency['vulnerabilities']:
+                # Get vulnerability data from JSON and setup variables
+                for vulnerability in dependency['vulnerabilities']:
+                    vuln_id_from_tool = title = description = cvss = cvssVector = cve = cvss_version = cwe = references = severity = None
+                    if "id" in vulnerability:
+                        vuln_id_from_tool = vulnerability["id"]
+                    if 'title' in vulnerability:
+                        title = vulnerability['title']
+                        # Find CWE in title in form "CWE-****"
+                        cwe_find = re.findall(r"^CWE-[0-9]{1,4}", title)
+                        if cwe_find:
+                            cwe = int(cwe_find[0][4:])
+                            # title = title.split(":")[1][1:] Unsure if AuditJS can specify both CWE and CVE at the same time
+                    if 'description' in vulnerability:
+                        description = vulnerability['description']
+                    if 'cvssScore' in vulnerability:
+                        cvss = vulnerability['cvssScore']
+                    # CVSS Vector not always given
+                    if 'cvssVector' in vulnerability:
+                        # Find if CVSS Version is specfied
+                        cvss_version = re.findall(r"^CVSS:[0-9]", vulnerability['cvssVector'])
+                        if cvss_version:
+                            severity = self.get_severity(cvss, vulnerability['cvssVector'][5:8])
+                            # If it's version 3.0 and above
+                            if vulnerability['cvssVector'][5] == "3":
+                                cvssVector = vulnerability['cvssVector'][9:]
+                                description = description + "\n\nCVSS V3 Vector: " + cvssVector
+                        else:
+                            severity = self.get_severity(cvss)
+                            description = description + "\n\nCVSS Vector: " + vulnerability['cvssVector']
+                    else:
+                        # Calculated based on CVSS V2
+                        severity = self.get_severity(cvss)
+                    if 'cve' in vulnerability:
+                        cve = vulnerability['cve']
+                        # title = title.split(":")[1][1:]
+                    if 'reference' in vulnerability:
+                        references = vulnerability['reference']
+
+                    finding = Finding(
+                        title=title,
+                        test=test,
+                        cve=cve,
+                        cwe=cwe,
+                        cvssv3=cvssVector,
+                        cvssv3_score=cvss,
```

Comment:
```suggestion
```
If you set the CVSS vector, the upper layers will manage the score automatically.

Reply:
Yes, that's right, but some findings in AuditJS do not have a CVSS Vector specified, so I prefer passing it anyway as the CVSS score is always there.

Reply:
Does DefectDojo compute severity automatically if I only pass the CVSS score and not vector ?
Therefore, I could delete the get_severity function (which I already simplified).

Reply:
For now , no. So the current version of the code is good.